temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowStoreImpl.java

```diff
@@ -351,7 +348,7 @@ public void sendQueryTask(
       task.setPreviousStartedEventId(previousStaredEventId);
       // it's not a real workflow task and the server sends 0 for startedEventId for such a workflow
       // task
-      task.setStartedEventId(startedEventId);
+      task.setStartedEventId(0);
```

Comment:
Can't say why  this was done since the comment in the code acknowledges it should be zero